src/mlpack/methods/random_forest/bootstrap.hpp

```diff
@@ -43,6 +45,153 @@ void Bootstrap(const MatType& dataset,
     bootstrapWeights = weights.cols(indices);
 }
 
+/**
+ * Default bootstrap strategy that uses the ::Bootstrap() function. 
+ */
+template<bool UseWeights>
```

Comment:
This is a minor comment, but maybe it would be simpler to make `UseWeights` a template parameter directly for the `Bootstrap()` function?  I don't see a particular reason why it needs to be held at the class level.